### ISSUE TITLE
Can't merge non text custom fields.

### ISSUE THREAD

**Issue by markalston** [Mailtrain-org/mailtrain#206]
Getting the following error in the logs everytime I try to merge a non-text custom field.  Text fields work fine. 

```
Apr 13 21:57:43 www node[26646]: /opt/mailtrain/node_modules/mysql/lib/protocol/Parser.js:79
Apr 13 21:57:43 www node[26646]:         throw err; // Rethrow non-MySQL errors
Apr 13 21:57:43 www node[26646]:         ^
Apr 13 21:57:43 www node[26646]: TypeError: string.replace is not a function
Apr 13 21:57:43 www node[26646]:     at Object.encode (/opt/mailtrain/node_modules/he/he.js:185:21)
Apr 13 21:57:43 www node[26646]:     at getValue (/opt/mailtrain/lib/tools.js:189:32)
Apr 13 21:57:43 www node[26646]:     at message.replace (/opt/mailtrain/lib/tools.js:198:22)
Apr 13 21:57:43 www node[26646]:     at RegExp.[Symbol.replace] (native)
Apr 13 21:57:43 www node[26646]:     at RegExp.[Symbol.replace] (native)
Apr 13 21:57:43 www node[26646]:     at String.replace (native)
Apr 13 21:57:43 www node[26646]:     at Object.formatMessage (/opt/mailtrain/lib/tools.js:196:20)
Apr 13 21:57:43 www node[26646]:     at render (/opt/mailtrain/routes/archive.js:70:65)
Apr 13 21:57:43 www node[26646]:     at renderHtml (/opt/mailtrain/routes/archive.js:90:33)
Apr 13 21:57:43 www node[26646]:     at renderAndShow (/opt/mailtrain/routes/archive.js:96:40)
Apr 13 21:57:43 www systemd[1]: mailtrain.service: Main process exited, code=exited, status=1/FAILURE
Apr 13 21:57:43 www systemd[1]: mailtrain.service: Unit entered failed state.
Apr 13 21:57:43 www systemd[1]: mailtrain.service: Failed with result 'exit-code'.
```
If I use a custom field with type text it works fine but any custom field set to number or date fails.  I have tested and this is independent of editor used.

I have a workaround by importing the same data into two columns. One set to numeric so that I can use it in segments and the other as text so that I can merge it.  Not an ideal solution however.

Is there some trick I am missing for using non text type custom fields in a merge?

**Comment by andris9** [comment]
If it throws an error like that then it is always a bug. I'll look into it

**Comment by markalston** [comment]
The following fix in lib/tools.js makes it work for me.  The downside is that dates are formatted like " Wed Apr 05 2017 18:00:00 GMT-0600 (MDT) "   but at least they don't crash mailtrain.
```
    let getValue = key => {
        key = (key || '').toString().toUpperCase().trim();
        if (links.hasOwnProperty(key)) {
            return links[key];
        }
        if (subscription.mergeTags.hasOwnProperty(key)) {
            return isHTML ? he.encode((subscription.mergeTags[key] || ''), {
                useNamedReferences: true
            }) : subscription.mergeTags[key].toSring();
        }
        return false;
    };
```

**Comment by witzig** [comment]
@markalston, thanks for reporting. Merge tags for date fields, and number fields with value 0, should now render correctly. Fixed with 466bf22.
